mlir/lib/Conversion/ArithToSPIRV/ArithToSPIRV.cpp

```diff
@@ -229,16 +230,29 @@ struct ConstantCompositeOpPattern final
     if (!srcType || srcType.getNumElements() == 1)
       return failure();
 
-    // arith.constant should only have vector or tenor types.
+    // arith.constant should only have vector or tensor types.
     assert((isa<VectorType, RankedTensorType>(srcType)));
```

Comment:
Is this the only supported case or is this the only possible case? Can you clarify this in the comment? If this is the only supported case, let's emit a failure with a nice error message instead.

Reply:
Good questions, there are TODOs all over MLIR to disallow other types, but at the moment almost everything is allowed. I emit a failure now.

Reply:
Sounds good to me.